build/nix/buildCurrent.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# This script builds the previous version and uses it to build the current version
+clear
+echo '*************************'
+echo Rebuilding Previous
+# The "target" directory is used for compiled versions of the program
+mkdir -p ../../target/previous
+rm -rf ../../target/previous/*
+# Build the old program into target/previous
+clang++ ../../translated/previous/*.cpp -o ../../target/previous/Program -std=c++14 -Xclang -flto-visibility-public-std
+echo ''
+echo '*************************'
+echo Compiling Current
+# The "translated" directory is where we store working versions of the program
+# Empty the "current" directory so we can fill it with the new build
+mkdir -p ../../translated/current
+rm -rf ../../translated/current/*
+source_files=`ls -d ../../src/*.ad ../../src/**/*.ad`
+# Use the previous version of the compiler to compile the new version into translated/current
+../../target/previous/Program $source_files -o ../../translated/current/Program.cpp -r ../../src/RuntimeLibrary.cpp -r ../../src/RuntimeLibrary.h
+
+# Build the new program into target/current for testing
+mkdir -p ../../target/current
+rm -rf ../../target/current/*
+clang++ ../../translated/current/*.cpp -o ../../target/current/Program -std=c++14 -Xclang -flto-visibility-public-std
```

Comment:
remove `-Xclang -flto-visibility-public-std` if it doesn't cause the build to break.  Pretty sure this is a windows specific issue  (note this in other scripts as well)